tests/functional/test_waiter_config.py

```diff
@@ -11,16 +11,94 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 import jmespath
+from jsonschema import Draft4Validator
 
 import botocore.session
+from botocore.exceptions import UnknownServiceError
 from botocore.utils import ArgumentGenerator
 
 
+WAITER_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "version": {"type": "number"},
+        "waiters": {
+            "type": "object",
+            "additionalProperties": {
+                "type": "object",
+                "properties": {
+                    "type": {
+                        "type": "string",
+                        "enum": ["api"]
+                    },
+                    "operation": {"type": "string"},
+                    "description": {"type": "string"},
+                    "delay": {
+                        "type": "number",
+                        "minimum": 0,
+                    },
+                    "maxAttempts": {
+                        "type": "integer",
+                        "minimum": 1
+                    },
+                    "acceptors": {
+                        "type": "array",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "state": {
+                                    "type": "string",
+                                    "enum": ["success", "retry", "failure"]
+                                },
+                                "matcher": {
+                                    "type": "string",
+                                    "enum": [
+                                        "path", "pathAll", "pathAny",
+                                        "status", "error"
+                                    ]
+                                },
+                                "argument": {"type": "string"},
```

Comment:
Not sure if it is possible in JSON schema, but is it possible to mark argument as required if matcher's value is `path`, `pathAll` or `pathAny`? Don't think it is a big deal as we have tests later on that assume argument is existing if any of those three matchers are used.

Reply:
ah yes

Reply:
No, conditional requirements aren't supported.